crates/sui-types/src/gas.rs

```diff
@@ -301,13 +301,13 @@ impl<'a> SuiGasStatus<'a> {
 }
 
 /// Check whether the given gas_object and gas_budget is legit:
-/// 1. If the gas object is owned.
+/// 1. If the gas object has an address owner.
 /// 2. If it's enough to pay the flat minimum transaction fee
 /// 3. If it's less than the max gas budget allowed
 /// 4. If the gas_object actually has enough balance to pay for the budget.
 pub fn check_gas_balance(gas_object: &Object, gas_budget: u64, gas_price: u64) -> SuiResult {
     ok_or_gas_error!(
-        gas_object.is_owned(),
+        matches!(gas_object.owner, Owner::AddressOwner(_)),
```

Comment:
@lxfind how should I test this change?

Reply:
Added a test

Reply:
Note though, this means the gas object cannot be a child object of _any_ kind. In my test, I made the parent object just a normal owned object.

We could make it just no quasi-shared objects, but that would be a much more involved check (since you would have to load the parent object, I think)

Reply:
Yeah I think that's perfectly fine. I don't see a need to use child object to pay gas